File: src/main/java/ru/arc/invest/BusinessLocation.java
package ru.arc.invest;

import com.sk89q.worldedit.bukkit.BukkitAdapter;
import com.sk89q.worldguard.WorldGuard;
import com.sk89q.worldguard.protection.regions.ProtectedRegion;
import com.sk89q.worldguard.protection.regions.RegionContainer;
import lombok.RequiredArgsConstructor;
import org.bukkit.Bukkit;
import org.bukkit.World;
import org.bukkit.configuration.serialization.ConfigurationSerializable;
import org.jetbrains.annotations.NotNull;

import java.util.Map;

@RequiredArgsConstructor
public class BusinessLocation implements ConfigurationSerializable {

    final String worldName;
    final String regionName;

    ProtectedRegion region;


    public void init(){
        World world = Bukkit.getWorld(worldName);
        if(world == null){
            System.out.println("World "+worldName+" was not found!");
            return;
        }

        RegionContainer regionContainer = WorldGuard.getInstance().getPlatform().getRegionContainer();
        this.region = regionContainer.get(BukkitAdapter.adapt(world)).getRegion(regionName);

        if (region == null) {
            System.out.print("No region named " + regionName + " in world " + worldName);
        }
    }

    public static BusinessLocation deserialize(Map<String, Object> map){
        String worldName = (String) map.get("world-name");
        String regionName = (String) map.get("region-name");

        BusinessLocation businessLocation = new BusinessLocation(worldName, regionName);
        return businessLocation;
    }

    @Override
    public @NotNull Map<String, Object> serialize() {
        return null;
    }
}

File: src/main/kotlin/ru/arc/invest/BusinessLocation.kt
package ru.arc.invest

import com.sk89q.worldedit.bukkit.BukkitAdapter
import com.sk89q.worldguard.WorldGuard
import com.sk89q.worldguard.protection.regions.ProtectedRegion
import org.bukkit.Bukkit
import org.bukkit.configuration.serialization.ConfigurationSerializable

class BusinessLocation(
    private val worldName: String,
    private val regionName: String,
) : ConfigurationSerializable {

    var region: ProtectedRegion? = null
        private set

    fun init() {
        val world = Bukkit.getWorld(worldName)
        if (world == null) {
            println("World $worldName was not found!")
            return
        }
        val regionContainer = WorldGuard.getInstance().platform.regionContainer
        region = regionContainer.get(BukkitAdapter.adapt(world))?.getRegion(regionName)
        if (region == null) {
            println("No region named $regionName in world $worldName")
        }
    }

    override fun serialize(): Map<String, Any> = mapOf(
        "world-name" to worldName,
        "region-name" to regionName,
    )

    companion object {
        @JvmStatic
        fun deserialize(map: Map<String, Any>): BusinessLocation {
            val worldName = map["world-name"] as String
            val regionName = map["region-name"] as String
            return BusinessLocation(worldName, regionName)
        }
    }
}
